### PR TITLE
Replace host OS detection code to deal with Java >9 versions

### PR DESCRIPTION
**Pull request by ggalmazor**
Addresses https://github.com/opendatakit/aggregate/issues/317

This PR replaces the obsolete Apache Commons method with an up-to-date, unit tested (and battle-tested) host OS detection algorithm we use in Briefcase.

### CODE REVIEW

File: AggregateAppEngineUpdater/build.gradle [hunk 1]
```diff
@@ -23,6 +23,11 @@ dependencies {
   compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.4'
```
Comment (yanokwa) [inline AggregateAppEngineUpdater/build.gradle]: You can drop this library. That'll save us some space in the Jar.

File: AggregateAppEngineUpdater/src/test/java/org/opendatakit/appengine/updater/HostTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package org.opendatakit.appengine.updater;/*
```
Comment (lognaturel) [inline AggregateAppEngineUpdater/src/test/java/org/opendatakit/appengine/updater/HostTest.java]: 😭 I hate to be nit-picky but I think this package declaration should go after the header!

**Review by yanokwa — CHANGES_REQUESTED** [review]

**Review by lognaturel — APPROVED** [review]

**Comment by yanokwa** [comment]
@ggalmazor Before we merge this, can you explain a bit about why we don't just upgrade Apache Commons? I know the Briefcase code is used in the field, but it's definitely not as well tested as Apache Commons.

File: AggregateAppEngineUpdater/src/test/java/org/opendatakit/appengine/updater/HostTest.java (hunk 2 above)
Comment (ggalmazor) [reply AggregateAppEngineUpdater/src/test/java/org/opendatakit/appengine/updater/HostTest.java]: I was trying to make a surgical change, but I guess you're right :)

**Comment by ggalmazor** [comment]
@yanokwa sure! Mainly two reasons:
- Apache Commons Lang is _huge_ and we just want to know if the host is Windows. It feels like an overkill.
- The actual test for the host detection code in Apache Commons Lang is this:
  ```java
  @Test
  public void testGetHostName() {
      final String hostName = SystemUtils.getHostName();
      final String expected = SystemUtils.IS_OS_WINDOWS ? System.getenv("COMPUTERNAME") : System.getenv("HOSTNAME");
      assertEquals(expected, hostName);
  }
  ```

  This test is in a class with a comment saying "Only limited testing can be performed". ¯\\\_(ツ)_/¯

  Our tests are more complete because they cover different host scenarios.

File: AggregateAppEngineUpdater/src/test/java/org/opendatakit/appengine/updater/HostTest.java (hunk 2 above)
Comment (ggalmazor) [reply AggregateAppEngineUpdater/src/test/java/org/opendatakit/appengine/updater/HostTest.java]: Thanks for spotting that! There was some problem while copy&pasting

**Review by yanokwa — APPROVED** [review]

**Comment by yanokwa** [comment]
Thanks for the explanations. The last commit cut the file size in half, so that's a nice win.

**Pull request merged**